src/Services/Notebook/NotebooksService.cs

```diff
@@ -0,0 +1,239 @@
+
+				await _db.SaveChangesAsync();
+
+				_logger.LogInformation(
+					"Saved Notebook to databse with  name: '{Name}' with id: {id}",
+					data.Id,
+					data.Name
+				);
+
+				transaction.Commit();
+
+				return new Result<NotebookDto>
+				{
+					Message = "Inserted",
+					IsSuccssfull = true,
+					Data = new NotebookDto
+					{
+						Id = data.Id,
+						Name = data.Name,
+						CreateDate = data.CreateDate,
+						Category = data.Category,
+						Color = data.Color
+					}
+				};
+
+			}
+			catch (Exception ex)
+			{
+				_logger.LogError("Failed to save Notebook because of exception: '{Message}'", ex.Message);
+
+				transaction.Rollback();
+
+				return new Result<NotebookDto>
+				{
+					Message = "Error while trying to save Notebook due to technical reason with code: " + ex.HResult,
+					IsSuccssfull = false,
+				};
+			}
+		}
+		public async Task<Result<NotebookDto>> Get(int id)
+		{
+			var data = await _db.Notebooks.FindAsync(id);
+
+			if (data == null)
+			{
+				return new Result<NotebookDto>
+				{
+					Message = "Notebook not found",
+					IsSuccssfull = false,
+				};
+			}
+
+			return new Result<NotebookDto>
+			{
+				IsSuccssfull = true,
+				Data = new NotebookDto
+				{
+					Id = data.Id,
+					Name = data.Name,
+					CreateDate = data.CreateDate,
+					Category = data.Category,
+					Color = data.Color
+				}
+			};
+		}
+
+		public async Task<Result> Delete(int id)
+		{
+			var transaction = _db.Database.BeginTransaction();
+
+			try
+			{
+				var notebook = await _db.Notebooks.FindAsync(id);
+				if (notebook == null)
+				{
+					return new Result
+					{
+						IsSuccssfull = false,
+						Message = "notebook not found",
+					};
+				}
+				_db.Notebooks.Remove(notebook);
+
+				await _db.SaveChangesAsync();
+
+				_logger.LogInformation(
+					"Deleted notebook from database with name '{Name}' and id: {Id}",
+					notebook.Name,
+					notebook.Id
+				);
+
+				transaction.Commit();
+
+				return new Result
+				{
+					IsSuccssfull = true,
+					Message = "Deleted from database",
+				};
+			}
+			catch (Exception ex)
+			{
+				_logger.LogError("Failed to delete notebook because of exception: '{Message}'", ex.Message);
+
+				transaction.Rollback();
+
+				return new Result
+				{
+					IsSuccssfull = false,
+					Message = "Error while trying to delete the notebook due to technical reason with code: " + ex.HResult,
+				};
+			}
+		}
+		public async Task<Result<NotebookDto>> Update(int id, NotebookDto notebook)
+		{
+			var transaction = _db.Database.BeginTransaction();
+
+			try
+			{
+				var entity = await _db.Notebooks.FindAsync(id);
+
+				if (entity == null)
+				{
+					return new Result<NotebookDto>
+					{
+						IsSuccssfull = false,
+						Message = "notebook not found",
+					};
+				}
+				entity.Name = notebook.Name;
+				entity.CreateDate = notebook.CreateDate;
+				entity.Color = notebook.Color;
+				entity.Category = notebook.Category;
+
+				_logger.LogInformation(
```

Comment:
please move the log directly under if the null so its logged immediately before doing any changes to the entity